cohortextractor/definition/base.py

```diff
@@ -0,0 +1,4 @@
+from weakref import WeakSet
```

Comment:
I'm not really sure if this is necessary/sensible or not; I was basing it on how Django registers admin sites, but maybe we don't care too much about it hanging around when everything's run in docker containers.

Reply:
My best guess is that this is _probably_ wrong -- couldn't the cohort objects could get garbage collected between the module being loaded (triggering the call to `register()`) and the cohort extractor actually accessing the definition?

Reply:
(And, as you suggest, I don't think we need to worry about these things hanging around since the whole process is fairly shortlived.)

Reply:
:+1:  I am a bit hazy on garbage collection, so definitely not committed to it.  I'll take it out

Reply:
While I was doing that, I realised that we can (and did, in the tests) have more than one cohort registered.  Last commit has a slight modification of the registry to allow tests to reset it, and to assert (for now, at least) that there's only one cohort registered.